docs/apps/lnw/es2k/es2k-lnw-underlay-ecmp.md

```diff
@@ -0,0 +1,456 @@
+
+```bash
+# VM1 configuration
+telnet <VM1 IP> <VM1 port>
+ip link add link <Netdev connected to VF1> name <Netdev connected to VF1>.10 type vlan id 10
+ip addr add 101.0.0.1/24 dev <Netdev connected to VF1>.10
+ifconfig <Netdev connected to VF> up
+ifconfig <Netdev connected to VF>.10 up
+```
+
+Option 2: If we are unable to spawn VM's on top of the VF's, we can leverage kernel network namespaces.
+Move each VF to a network namespace and assign IP addresses.
+Example: Below config is provided for one VM, and considering each namespace is in one VLAN. Extend this to other namespaces.
+
+```bash
+ip netns add VM0
+ip link set <VF1 port> netns VM0
+ip netns exec VM0 ip link add link <VF1 port> name <VF1 port>.10 type vlan id 10
+ip netns exec VM0 ip addr add 101.0.0.1/24 dev <VF1 port>.10
+ip netns exec VM0  ifconfig <VF1 port> up
+ip netns exec VM0  ifconfig <VF1 port>.10 up
+```
+
+### Configure rules for mapping between Overlay VF and ACC port representor
+
+Configure rules to send overlay packets from a VM to its respective port representors.
+
+Refer above port mapping for overlay VF to ACC port representor mapping. Here sample commands are shown for a single overlay network, configure similar mapping for remaining VFs.
+
+Example:
+
+- Overlay VF1 has a VSI value 27
+- Corresponding port representor VSI value 9
+- If a VSI is used as an action, add an offset of 16 to the VSI value
+
+```bash
+# Create a source port for an overlay VF (VSI-27). Source port action can be any value.
+# For simplicity add 16 to VSI ID.
+ p4rt-ctl add-entry br0 linux_networking_control.tx_source_port_v4 \
+     "vmeta.common.vsi=27,zero_padding=0,action=linux_networking_control.set_source_port(43)"
+
+# Create a mapping between overlay VF (VSI-27/source port-43) and ACC port representor (VSI-9)
+ p4rt-ctl add-entry br0 linux_networking_control.source_port_to_pr_map \
+     "user_meta.cmeta.source_port=43,zero_padding=0,action=linux_networking_control.fwd_to_vsi(25)"
+
+ p4rt-ctl add-entry br0 linux_networking_control.tx_acc_vsi \
+     "vmeta.common.vsi=9,zero_padding=0,action=linux_networking_control.l2_fwd_and_bypass_bridge(43)"
+
+# Create a mapping for traffic to flow between VSIs (VSI-27/source port-43) and (VSI-9)
+ p4rt-ctl add-entry br0 linux_networking_control.vsi_to_vsi_loopback \
+     "vmeta.common.vsi=9,target_vsi=27,action=linux_networking_control.fwd_to_vsi(43)"
+
+ p4rt-ctl add-entry br0 linux_networking_control.vsi_to_vsi_loopback \
+     "vmeta.common.vsi=27,target_vsi=9,action=linux_networking_control.fwd_to_vsi(25)"
+
+```
+
+### Configure rules for mapping between Physical port and ACC port representor
+
+Configure rules to send ingress packets from a physical port to its respective port representors.
+
+Refer above port mapping for physical port to ACC port representor mapping. Here sample commands are shown for a single physical port, configure similar mapping for remaining physical ports.
```

Comment:
```suggestion
Refer to above port mapping for physical port to ACC port representor mapping. Here, sample commands are shown for a single physical port. Configure similar mapping for remaining physical ports.
```

1. _Refer to_, not _refer_. They have different meanings.
  - "Refer to X" instructs the reader to look elsewhere for information.
  - "Refer X to Y" means to tell person or persons X to go see person Y (or an example of _metonymy_, "a figure of speech that consists of the use of the name of one object or concept for that of another to which it is related, or of which it is a part").
2. Comma following disjunct adverb.
3. Comma splice. Start a new sentence.